tests/util/test_logcontext.py

```diff
@@ -109,7 +109,9 @@ def testfunc():
     def test_run_in_background_with_coroutine(self):
         async def testfunc():
             self._check_test_key("one")
-            d = Clock(reactor).sleep(0)
+            # Theoretically the ensureDeferred shouldn't be necessary here, but
+            # it forces trial to fire the awaitable.
```

Comment:
Sorry, can you expand on this a bit? I don't really understand what you mean by "fire"?

Reply:
I think in order to get Twisted to actually run the sleep call it needs to be wrapped in `ensureDeferred`?

That doesn't really seem right though since the `run_in_background` call in `_test_run_in_background` should handle that.

Without this change it was hanging on the `await d` call below.